modules/stochastic_tools/test/tests/surrogates/libtorch_nn/tests

```diff
@@ -0,0 +1,60 @@
+[Tests]
+  issues = "#19571"
+  design = "LibtorchANNTrainer.md LibtorchANNSurrogate.md"
+
+  [train_and_evaluate]
+    requirement = 'The system shall be able to train and evaluate a libtorch-based neural network in the same input file.'
+    type = CSVDiff
+    input = train_and_evaluate.i
+    allow_test_objects = true
+    cli_args = "Trainers/train/read_from_file=false"
+    csvdiff = 'train_and_evaluate_out_results_0002.csv'
+    libtorch = true
+  []
+  [train]
+    requirement = 'The system shall be able to train a libtorch-based neural network.'
+    type = CheckFiles
+    input = train.i
+    check_files = 'train_out_train.rd mynet.pt'
+    cli_args = "Trainers/train/read_from_file=false"
+    allow_test_objects = true
+    libtorch = true
+  []
+  [evaluate]
+    requirement = 'The system shall be able to evaluate a previously trained, libtorch-based neural network.'
+    type = CSVDiff
+    input = evaluate.i
+    allow_test_objects = true
+    csvdiff = 'evaluate_out_results_0002.csv'
+    prereq = train
+    libtorch = true
+  []
+  [retrain]
+    requirement = 'The system shall be able to retrain a pretrained and saved libtorch-based neural network.'
+    type = CSVDiff
+    input = retrain.i
+    allow_test_objects = true
+    cli_args = "Trainers/train/read_from_file=true"
+    csvdiff = 'retrain_out_results_0002.csv'
+    prereq = evaluate
+    libtorch = true
+  []
+  [trainer-no-torch-error]
+    requirement = 'The system shall be able to throw an error if the user requires libtorch-based objects without installing libtorch.'
+    type = RunException
+    input = train.i
+    allow_test_objects = true
+    expect_err = "PyTorch C\+\+ API \(libtorch\) must be installed to use this object, see https://mooseframework.inl.gov/modules/stochastic_tools/install_pytorch.html for instruction."
+    libtorch = false
```

Comment:
This flag didn't seem to work on my mac. It still ran the test (with libtorch) which threw an error.

Reply:
Oh just saw I got a bunch of these:
```

Parser Errors:
--------------------------------------------------------------------------------------------------------------
/Users/prinzm/tmp/moose/modules/stochastic_tools/test/tests/surrogates/libtorch_nn/tests:12: unused parameter "libtorch"
/Users/prinzm/tmp/moose/modules/stochastic_tools/test/tests/surrogates/libtorch_nn/tests:21: unused parameter "libtorch"
/Users/prinzm/tmp/moose/modules/stochastic_tools/test/tests/surrogates/libtorch_nn/tests:30: unused parameter "libtorch"
/Users/prinzm/tmp/moose/modules/stochastic_tools/test/tests/surrogates/libtorch_nn/tests:40: unused parameter "libtorch"
/Users/prinzm/tmp/moose/modules/stochastic_tools/test/tests/surrogates/libtorch_nn/tests:48: unused parameter "libtorch"
/Users/prinzm/tmp/moose/modules/stochastic_tools/test/tests/surrogates/libtorch_nn/tests:57: unused parameter "libtorch"
```

Reply:
Did you try a clean installation or rebased on my branch? I am confused a little with the unused parameter errors. 

Could you check python/TestHarness/testers/Tester.py if you have libtorch defined? It should be there in my repo but I remember having a conflict there with something else while rebasing at some point.

Could you give me a bit more details on this issue so that I can figure out what fell through?

Reply:
My PYTHONPATH was pointing to another moose I had :(. Sorry!